cpp/src/arrow/compute/kernels/scalar_nested_test.cc

```diff
@@ -39,6 +39,46 @@ TEST(TestScalarNested, ListValueLength) {
   }
 }
 
+TEST(TestScalarNested, ListElement) {
+  for (auto ty : {list(int16()), fixed_size_list(uint8(), 2)}) {
+    auto input = ArrayFromJSON(ty, "[[8, 5], [74, 92], [7, 4]]");
+    auto out_ty = ty->id() == Type::LIST ? int16() : uint8();
+    auto expected = ArrayFromJSON(out_ty, "[5, 92, 4]");
```

Comment:
Add invalid tests:
* Index out-of-range
* Arrays of different lengths

Reply:
Thanks! Good catch!